### ISSUE TITLE
`theme download` should store checksums of each individually downloaded file for faster comparisson

### ISSUE THREAD

**Issue by vfonic** [Shopify/themekit#835]
**Is your feature request related to a problem? Please describe.**

This is related to #801 (theme download gets stuck at 97%).
The issue is that `theme download` never reaches 100%.

**Describe the solution you'd like**

I noticed that in the newer versions of themekit, themekit smartly compares local and remote files,
making it much faster on subsequent runs.
I'd love to be able to use this feature.

**Describe alternatives you've considered**

I've considered using v1.0.2, but v1.0.2 downloads all files every time I run `theme download`.

**Comment by tanema** [comment]
I am sorry, is this an issue or is this just you trying to add to an already existing issue?

**Comment by vfonic** [comment]
To me this seemed like a separate issue.

I don't know the internals of themekit, but what I noticed recently is that newer versions of themekit somehow magically remember which files are the same and don't need to be downloaded again.

Running `theme download` twice on v1.0.2 will run the full theme download twice.
Running `theme download` twice on v1.1.2 will run the full theme download once and will just download couple of files the second time.

Is this related to `theme download` stalls at 96%? I don't think so.

Please also notice that this is a feature request, and not a bug report.

**Comment by tanema** [comment]
This is directly related to that functionality, and the functionality already exists and is implemented.

**Comment by vfonic** [comment]
@tanema from my experience using themekit, this is what seems to happen after you run `theme download`:

1. The files are downloaded in parallel
2. Once _all_ of the files are downloaded, the checksums are stored

The workflow I'm suggesting is:
1. The files are downloaded in parallel
2. As each file is downloaded, its checksum is stored immediately.

Perhaps I don't fully understand how this works, especially since I don't see any local changes if `theme download` successfully finishes (I have no clue where/how the checksums are stored). This would partially fix the other issue, as subsequent `theme download`s might actually pass as they will download less files. The main issue of #801 will still not be addressed (initial download).

**Comment by tanema** [comment]
Themekit does not store anything other than files. Themekit needs a file on the filesystem to check the content of.
